### ISSUE TITLE
list_blobs does not raise the exception but returns the exception when the container does not exist

### ISSUE THREAD

**Issue by AbelHu** [Azure/azure-storage-ruby#37]
When the container does not exist, list_blob should raise an exception. But in v0.10.2, it returns the exception.

https://github.com/Azure/azure-storage-ruby/blob/master/lib/azure/storage/blob/container.rb#L551

```
def list_blobs(name, options={})
      # Call
      response = call(:get, uri)

      # Result
      if response.success?
        Serialization.blob_enumeration_results_from_xml(response.body)
      else
        response.exception
      end
end
```

**Comment by yaxia** [comment]
Thanks for reporting this issue. I will investigate and make the API behaviors consistent.

**Comment by yaxia** [comment]
It is fixed in 0.11.4 (with the latest azure-core). Please take a try. Thanks.
